packages/kbn-tinymath/test/library.test.js

```diff
@@ -31,96 +43,140 @@ describe('Parser', () => {
-      expect(check("'foo bar '")).toThrow('but "\'" found');
-      expect(check("'0foo'")).toThrow('but "\'" found');
-      /* eslint-enable prettier/prettier */
+      expect(parse('"_foo bar"')).toEqual(variableEqual('_foo bar'));
+      expect(parse('"@foo bar"')).toEqual(variableEqual('@foo bar'));
+      expect(parse('".foo bar"')).toEqual(variableEqual('.foo bar'));
+      expect(parse('"-foo bar"')).toEqual(variableEqual('-foo bar'));
+      expect(parse('"_foo0 bar1"')).toEqual(variableEqual('_foo0 bar1'));
+      expect(parse('"@foo0 bar1"')).toEqual(variableEqual('@foo0 bar1'));
+      expect(parse('".foo0 bar1"')).toEqual(variableEqual('.foo0 bar1'));
+      expect(parse('"-foo0 bar1"')).toEqual(variableEqual('-foo0 bar1'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
+      expect(parse('" foo bar"')).toEqual(variableEqual(' foo bar'));
+      expect(parse('"foo bar "')).toEqual(variableEqual('foo bar '));
+      expect(parse('"0foo"')).toEqual(variableEqual('0foo'));
     });
   });
 
   describe('Functions', () => {
     it('no arguments', () => {
-      expect(parse('foo()')).toEqual({ name: 'foo', args: [] });
+      expect(parse('foo()')).toEqual(functionEqual('foo', []));
     });
 
     it('arguments', () => {
-      expect(parse('foo(5,10)')).toEqual({ name: 'foo', args: [5, 10] });
+      expect(parse('foo(5,10)')).toEqual(functionEqual('foo', [5, 10]));
     });
 
     it('arguments with strings', () => {
-      expect(parse('foo("string with spaces")')).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
+      expect(parse('foo("string with spaces")')).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
 
-      /* eslint-disable prettier/prettier */
-      expect(parse("foo('string with spaces')")).toEqual({
-        name: 'foo',
-        args: ['string with spaces'],
-      });
-      /* eslint-enable prettier/prettier */
+      expect(parse("foo('string with spaces')")).toEqual(
+        functionEqual('foo', [variableEqual('string with spaces')])
+      );
+    });
+
+    it('named only', () => {
+      expect(parse('foo(q=10)')).toEqual(functionEqual('foo', [namedArgumentEqual('q', 10)]));
+    });
+
+    it('named argument is numeric', () => {
+      expect(parse('foo(q=10.1234e5)')).toEqual(
+        functionEqual('foo', [namedArgumentEqual('q', 10.1234e5)])
+      );
+    });
+
+    it('named and positional', () => {
+      expect(parse('foo(ref, q="bar")')).toEqual(
+        functionEqual('foo', [variableEqual('ref'), namedArgumentEqual('q', 'bar')])
+      );
+    });
+
+    it('numerically named', () => {
+      expect(() => parse('foo(1=2)')).toThrow('but "(" found');
+    });
+
+    it('multiple named', () => {
+      expect(parse('foo(q_param="bar", offset-type="1d")')).toEqual(
```

Comment:
Would it be possible to support quotes around parameter names as well?

Reply:
That sounds like asking for quotes around function names: sure, we could do it, but I don't think it's what users would expect to see.

Reply:
Technically yes, but given the example in that line, I see this version not so exotic:
`foo(q_param="bar", "offset-type"="1d")`

Reply:
@dej611 I've never seen a grammar like that- have you? I was modeling this on something like python named arguments, or even what Timelion does.

Reply:
<details>
  <summary>Ruby?</summary>

  ```ruby
  def foo(my_bar: 'default')
    puts my_bar
  end

  foo # => 'default'
  foo('my_bar': 'baz') # => 'baz'
  ```

</details>

In Python it's discouraged to use the kebab-case for parameters, unless using [very hacky tricks](https://stackoverflow.com/a/24121330/2964675), so there's no much need. In the call above snakecase and kebabcase are used in a mixed way and are both valid (could not find a mainstream language that does support both cases as keyword arguments).

It's a minor thing, I understand if it's not implemented.

Reply:
Oh this makes sense now! I will remove `-` from the allowed characters in named arguments, and I think that will resolve all these issues.